Add Api Unavailable Message #92

eddiesarevalo opened the pull request:
## 🗒️ Summary
-Added a message that appears at the top of the app if the app can not connect to the api.
-A test is sent on startup of the app.
-Every time a call is made to the api the api is tested.
-The message has a link to the pds-operator page.

## ⚙️ Test Data and/or Report
Testing on start up:
Start up the app using `npm start`
Connect to pds-dev3 and refresh the app. There will not be an error at the top.
Disconnect from pds-dev3 then refresh the app. There will be an error message at the top. 

During Run:
Running search, creation, reservation while disconnected from pds-dev3 will show the error message at the top. Being connected will not show the error message.

Click on the PDS Operator link to make sure it links to the PDS operator page.

## ♻️ Related Issues
- fixes #92

tloubrieu-jpl reviewed (changes requested):
Thanks @eddiesarevalo , that looks good. I lie the red banner on top.
 But it does not disappear if the API is up again.

See the screenshot.
<img width="1447" alt="image" src="https://user-images.githubusercontent.com/60993872/163890943-5a1fe202-a875-44eb-859b-e19a4751ff89.png">

The test scenario is:
1. start to use the UI without the API
2. start the API
3. re-do the same reserve action

eddiesarevalo commented:
@tloubrieu-jpl Good catch! I wasn't setting the error message off when a successful connection was made. I was only setting if off on a successful search.

eddiesarevalo commented:
@tloubrieu-jpl I applied the fix for the odd styling bug after the app is built. It was occuring because the theme provider was missing. So every component would interfere with each other's styling. 

I also added some missing styling for the call to action buttons on the home menu.

The pds app bar has also been removed. Since it should be deployed separately from the doi-ui rather then be embedded into it. 

I added the testing instructions to the pull request description.
